scheduler/replica_scheduler.go

```diff
@@ -569,3 +570,55 @@ func (rcs *ReplicaScheduler) GetDiskSchedulingInfo(disk types.DiskSpec, diskStat
 	}
 	return info, nil
 }
+
+func (rcs *ReplicaScheduler) CheckReplicasSizeExpansion(v *longhorn.Volume, oldSize, newSize int64) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, "error while CheckReplicasSizeExpansion for volume %v", v.Name)
+	}()
+
+	replicas, err := rcs.ds.ListVolumeReplicas(v.Name)
+	if err != nil {
+		return err
+	}
+	diskIDToReplicaCount := map[string]int64{}
+	diskIDToDiskInfo := map[string]*DiskSchedulingInfo{}
+	for _, r := range replicas {
+		if r.Spec.NodeID == "" {
+			continue
+		}
+		node, err := rcs.ds.GetNode(r.Spec.NodeID)
+		if err != nil {
+			return err
+		}
+		diskSpec, diskStatus, ok := findDiskSpecAndDiskStatusInNode(r.Spec.DiskID, node)
+		if !ok {
+			return fmt.Errorf("cannot find the disk %v in node %v", r.Spec.DiskID, node.Name)
+		}
+		diskInfo, err := rcs.GetDiskSchedulingInfo(diskSpec, &diskStatus)
+		if err != nil {
+			return fmt.Errorf("failed to GetDiskSchedulingInfo %v", err)
+		}
+		diskIDToDiskInfo[r.Spec.DiskID] = diskInfo
+		diskIDToReplicaCount[r.Spec.DiskID] = diskIDToReplicaCount[r.Spec.DiskID] + 1
+	}
+
+	delSize := newSize - oldSize
```

Comment:
NIT: delSize could be deltaSize or deletionSize, I feel like this is good spot to be a little bit more descriptive :)

Reply:
`expandedSize` may be more precise.

Reply:
Thank you! Fixed